server/src/executor/command_handlers.rs

```diff
@@ -108,6 +110,36 @@ impl CommandHandler for SpawnCommandHandler {
     }
 }
 
+#[derive(Constructor)]
+pub struct RespawnCommandHandler {
+    player_id: u32,
+}
+
+impl CommandHandler for RespawnCommandHandler {
+    fn handle(
+        &self,
+        game_state: &mut GameState,
+        physics_state: &mut PhysicsState,
+        _: &mut dyn GameEventCollector,
+    ) -> HandlerResult {
+        // TODO: remove debug code, example usage
+        // if !command_on_cooldown(game_state, self.player_id, Command::Respawn) {
+        //     return Ok(());
+        // }
+        let ground_groups = InteractionGroups::new(1.into(), 1.into());
```

Comment:
this is kind of repeating the code of the spawn handler. Maybe it'll make more sense to just teleport the players to correct positions instead of creating the colliders, etc, again.

Reply:
That makes sense, I'll look into it